Update README and add MIGRATION_GUIDE (#230)
klaviyo/klaviyo-react-native-sdk: README.md

  - [In-App Forms](#in-app-forms)
    - [Prerequisites](#prerequisites-1)
    - [Setup](#setup-1)
      - [Registering for in-app forms](#registering-for-in-app-forms)
      - [App Session configuration](#app-session-configuration)

>       - [Unregistering from in-app forms](#unregistering-from-in-app-forms)

Author: lol @ the 34 usages currently in this README

[In-app forms](https://help.klaviyo.com/hc/en-us/articles/34567685177883) are messages displayed to mobile app users while they are actively using an app. You can create new in-app forms in a drag-and-drop editor in the Sign-Up Forms tab in Klaviyo. Follow the instructions in this section to integrate forms with your app. The SDK will display forms according to their targeting and behavior settings and collect delivery and engagement analytics automatically.

> Beginning with version 1.3.0, in-app forms support advanced targeting and segmentation. In your Klaviyo account, you can configure forms to target or exclude a specific list or segment, and the form will be displayed to (or hidden from) profiles included in the specified list or segment.

Reviewer: ```suggestion
Beginning with version 1.3.0, in-app forms support advanced targeting and segmentation. In your Klaviyo account, you can configure forms to target or exclude a specific list or segment, and the form will display
based on this.
```

but honestly I'm not a product person so I don't think my commentary holds much weight
